src/poetry/repositories/pypi_repository.py

```diff
@@ -243,12 +243,16 @@ def _get_release_info(
 
     def _get(self, endpoint: str) -> dict[str, Any] | None:
         try:
-            json_response = self.session.get(self._base_url + endpoint)
+            json_response = self.session.get(
+                self._base_url + endpoint, raise_for_status=False
```

Comment:
This changes the behaviour for all errors, some of which might plausibly contain a json body.  In that case the eventual failure is likely to come at some later and more confusing point.

Suggest that still want to raise_for_status on non-404 errors.

Reply:
We can get `raise_for_status` to accept a filter I guess. But it should also be noted that this is he behaviour for legacy anyway right now. Maybe logging the errors and not bailing out is a better compromise.

Reply:
```console
Creating new session for pypi.org
[urllib3.connectionpool] Starting new HTTPS connection (1): pypi.org:443
[urllib3.connectionpool] https://pypi.org:443 "GET /pypi/fooobaaar/json/ HTTP/1.1" 404 1979
Source (PyPI): No packages found for fooobaaar >=0.1.0,<0.2.0
Falling back to installed packages to discover metadata for fooobaaar
Found 0 compatible packages for fooobaaar
   1: fact: no versions of fooobaaar match >=0.1.0,<0.2.0
   1: conflict: no versions of fooobaaar match >=0.1.0,<0.2.0
```

This is what the logs look like with the fix.